### DISCUSSION TITLE
USB3 vs USB2 on the Pi 4B

### DISCUSSION THREAD

**Discussion by mpfl** [mcguirepr89/BirdNET-Pi#455] [Q&A] [answered]
After some initial success I've been having issues with my new BirdNET-Pi setup.

It consists of a Pi4B 4GB with a Logitech C270 webcam (you work with what you have!)

I had the webcam plugged into one of the blue USB3 ports. The instance would work well all day and then suddenly stop working properly. Rebooting, restarting services, replugging the webcam into the same port would work for a little while but then fail again quickly.

I looked at `dmesg` and saw it full of errors like this:

```
[  717.052897] usb 1-1.2: reset high-speed USB device number 3 using xhci_hcd
[  717.802124] hdmi-audio-codec hdmi-audio-codec.4.auto: ASoC: error at snd_soc_dai_startup on i2s-hifi: -19
[  717.802184]  MAI: soc_pcm_open() failed (-19)
[  717.802748] hdmi-audio-codec hdmi-audio-codec.4.auto: ASoC: error at snd_soc_dai_startup on i2s-hifi: -19
[  717.802783]  MAI: soc_pcm_open() failed (-19)
```

I swapped the webcam from the blue USB3 port to a black USB2 port and BirdNET-Pi seems to be working happily again, with these lines as the last entries in `dmesg`:

```
[  878.979036] usb 1-1.2: USB disconnect, device number 3
[  878.979433] usb 1-1.2: cannot submit urb (err = -19)
[  878.980068] usb 1-1.2: cannot submit urb 0, error -19: no device
[  882.439705] usb 1-1.4: new high-speed USB device number 4 using xhci_hcd
[  882.752965] usb 1-1.4: New USB device found, idVendor=046d, idProduct=0825, bcdDevice= 5.62
[  882.752986] usb 1-1.4: New USB device strings: Mfr=0, Product=2, SerialNumber=1
[  882.752994] usb 1-1.4: Product: C270 HD WEBCAM
[  882.753000] usb 1-1.4: SerialNumber: CE749660
[  882.754430] usb 1-1.4: Found UVC 1.00 device C270 HD WEBCAM (046d:0825)
[  882.948732] usb 1-1.4: Failed to query (GET_INFO) UVC control 3 on unit 1: -32 (exp. 1).
[  882.976177] input: C270 HD WEBCAM as /devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.4/1-1.4:1.0/input/input7
[  883.040195] usb 1-1.4: set resolution quirk: cval->res = 384
```

Is anyone more familiar with the quirks of Raspberry Pi USB? Do you have any tips for me?

**Comment by Clumpton** [comment]
The Pi4B is renowned for having problems with the USB3 ports interfering with WiFi and Bluetooth.
I have had to turn of both radio and BT and add external dongles (which work fine) on a couple of pi4 projects now.
Your problems come as no surprise.

**Comment by mpfl** [comment, marked as answer]
I continued to have the issue on USB2 ports as well, unfortunately.

But a couple of days ago I got a UGREEN USB audio adapter and a better microphone. Since replacing the webcam with one of those, the audio recording has been rock solid.

So I would hazard that it was the USB webcam that had the problem, not the Pi USB port.
